kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt

```diff
@@ -0,0 +1,374 @@
+
+            // Creates a dialog to choose a where clause
+            val alertDialog: AlertDialog.Builder = AlertDialog.Builder(this)
+            alertDialog.setTitle("AlertDialog")
+            val checkedItem = cqlQueryListPosition
+
+            alertDialog.setSingleChoiceItems(
+                cqlQueryList.toTypedArray(), checkedItem
+            ) { dialog, which ->
+
+                // Updates the selected where clause
+                cqlQueryListPosition = which
+                cqlFiltersLayoutBinding.cqlQueryTextView.text = cqlQueryList[which]
+
+                // Dismiss dialog
+                dialog.dismiss()
+            }
+
+            // Displays the where clause dialog
+            val alert: AlertDialog = alertDialog.create()
+            alert.show()
+        }
+
+        // Sets the view to the default value of max features (1000)
+        cqlFiltersLayoutBinding.maxFeaturesEditText.setText(maxFeatures.toString())
+
+        // Sets from date to Jun-13-2011 by default
+        cqlFiltersLayoutBinding.fromDatePicker.updateDate(2011, 5, 13)
+        // Sets to date to Jan-7-2012 by default
+        cqlFiltersLayoutBinding.toDatePicker.updateDate(2012, 0, 7)
+
+        // Sets up filters for the query when Apply is clicked.
+        cqlFiltersLayoutBinding.applyTv.setOnClickListener {
+
+            // Retrieves the max features
+            val maxFeaturesText =
+                cqlFiltersLayoutBinding.maxFeaturesEditText.text.toString()
+            maxFeatures = when {
+                maxFeaturesText == "" -> 1000
+                maxFeaturesText.toInt() <= 0 -> 1000
+                else -> maxFeaturesText.toInt()
+            }
+
+            // Retrieves if date filter is selected
+            isDateFilter = cqlFiltersLayoutBinding.dateSwitch.isChecked
+
+            // Retrieves from & to dates from the DatePicker
+            val fromDatePicker = cqlFiltersLayoutBinding.fromDatePicker
+            val toDatePicker = cqlFiltersLayoutBinding.toDatePicker
+            fromDate.set(fromDatePicker.year, fromDatePicker.month, fromDatePicker.dayOfMonth)
+            toDate.set(toDatePicker.year, toDatePicker.month, toDatePicker.dayOfMonth)
+
+            // Dismiss bottom sheet view
+            dialog.dismiss()
+
+            // Runs the query using the selected filters
+            runQuery()
+        }
+
+        // Dismiss bottom sheet view when cancel is clicked
+        cqlFiltersLayoutBinding.cancelTv.setOnClickListener { dialog.dismiss() }
+        dialog.setCancelable(false)
+
+        // Sets bottom sheet content view to layout
+        dialog.setContentView(cqlFiltersLayoutBinding.root)
+
+        // Displays bottom sheet view
+        dialog.show()
+    }
+
+    /**
+     * Resets views in bottomSheet to default values.
+     */
+    private fun resetBottomSheetValues() {
+        cqlQueryListPosition = 4
+        maxFeatures = 1000
+        isDateFilter = false
```

Comment:
hook this into the component: it might include an extra line of code to set the check box ticked to false.